drivers/digital-io/max22190/max22190.c

```diff
@@ -0,0 +1,517 @@
+				((datainput >> (length -2 - i)) & 0x01));
+
+		if ((tmp & 0x20) == 0x20)
+			crc_step = (uint8_t)(tmp ^ crc_poly);
+		else
+			crc_step = tmp;
+	}
+
+	return (uint8_t)(crc_step & 0x1F);
+}
+
+/**
+ * @brief Set filter delay, wite break detection and if the filter is used or
+ * 	  bypassed.
+ * @param desc - MAX22190 device descriptor.
+ * @param ch - The channel to which the filter corresponds(0 based).
+ * @param wbe - Wire Break detection state(disabled - 0, enabled - 1).
+ * @param fbp - Filter usage(used - 0, bypassed - 1).
+ * @param delay - Filter delay.
+ * @return 0 in case of success, negative error code otherwise.
+*/
+int max22190_filter_set(struct max22190_desc *desc, uint32_t ch, uint32_t wbe,
+			uint32_t fbp, enum max22190_delay delay)
+{
+	if (ch > MAX22190_CHANNELS - 1)
+		return -EINVAL;
+
+	return max22190_reg_write(desc, MAX22190_FILTER_IN_REG(ch),
+				  no_os_field_prep(MAX22190_WBE_MASK, wbe) |
+				  no_os_field_prep(MAX22190_FBP_MASK, fbp) |
+				  no_os_field_prep(MAX22190_DELAY_MASK, delay));
+}
+
+/**
+ * @brief Get filter delay, wite break detection and if the filter is used or
+ * 	  bypassed.
+ * @param desc - MAX22190 device descriptor.
+ * @param ch - The channel to which the filter corresponds(0 based).
+ * @param wbe - Wire Break detection state(disabled - 0, enabled - 1).
+ * @param fbp - Filter usage(used - 0, bypassed - 1).
+ * @param delay - Filter delay.
+ * @return 0 in case of success, negative error code otherwise.
+*/
+int max22190_filter_get(struct max22190_desc *desc, uint32_t ch, uint32_t *wbe,
+			uint32_t *fbp, enum max22190_delay *delay)
+{
+	int ret;
+	uint32_t read_byte;
+
+	if (ch > MAX22190_CHANNELS - 1)
+		return -EINVAL;
+
+	ret = max22190_reg_read(desc, MAX22190_FILTER_IN_REG(ch), &read_byte);
+	if (ret)
+		return ret;
+
+	*wbe = no_os_field_get(MAX22190_WBE_MASK, read_byte);
+	*fbp = no_os_field_get(MAX22190_FBP_MASK, read_byte);
+	*delay = no_os_field_get(MAX22190_DELAY_MASK, read_byte);
+
+	return 0;
+}
+
+/**
+ * @brief Switch selected channel's state to disabled or enabled.
+ * @param desc - MAX22190 device descriptor.
+ * @param ch - Channel to disable/enable(0 based).
+ * @param state - The state requested.
+ * @return 0 in case of success, negative error code otherwise.
+*/
+int max22190_chan_state(struct max22190_desc *desc, uint32_t ch,
+			enum max22190_ch_state state)
+{
+	int ret;
+
+	if (ch > MAX22190_CHANNELS - 1)
+		return -EINVAL;
+
+	ret = max22190_reg_update(desc, MAX22190_INPUT_EN_REG,
+				  MAX22190_CH_STATE_MASK(ch), state);
+	if (ret)
+		return ret;
+
+	desc->channels[ch] = state;
+
+	return 0;
+}
+
+/**
+ * @brief Get fault enable from the fault registers
+ * @param desc - MAX22190 device descriptor
+ * @param fault_enable - Specific fault enable to read
+ * @param enabled - State of the selected enable (0 for disabled, 1 for enabled)
+ * @return 0 in case of succes, negative error code otherwise
+*/
+int max22190_fault_enable_get(struct max22190_desc *desc,
```

Comment:
let's add similar APIs for other registers (e.g. config register, filters, etc)

Reply:
Config reg has 2 bits that can be added to fault and 1 for register, and I will do exactly that.